src/WriteTransaction.ts

```diff
@@ -35,81 +41,89 @@ export class WriteTransaction extends Transaction<WriteTransactionRequest, Write
       const req = new WriteTransactionRequest()
       req.setModelcreaterequest(createReq)
       this.client.onMessage((message: WriteTransactionReply) => {
-        resolve(message.hasModelcreatereply())
+        const reply = message.getModelcreatereply()
+        resolve(reply ? reply.toObject() : undefined)
       })
       super.setReject(reject)
       this.client.send(req)
     })
   }
 
   public async modelSave(values: any[]) {
-    return new Promise<boolean>((resolve, reject) => {
+    return new Promise<ModelSaveReply.AsObject>((resolve, reject) => {
       const saveReq = new ModelSaveRequest()
       const list: any[] = []
       values.forEach(v => {
-        v['ID'] = uuid.v4()
+        if (!v.hasOwnProperty('ID')) {
+          throw new Error('missing "ID" property')
```

Comment:
Actually, in this case we should just add the ID property but set it to "". The server will add an ID if not present.

Reply:
Cool, will do.